### PR TITLE
Fix for Solaris 10 pkg.install issue 59305 and tests

### LINKED ISSUE #59305: [BUG] solarispkg.install only supports string "True" for current_zone_only kwarg
**Description**

The [`solarispkg.install`](https://docs.saltproject.io/en/master/ref/modules/all/salt.modules.solarispkg.html#salt.modules.solarispkg.install) function documents the use of `current_zone_only=True`, but in fact it only tests for the _string value_ `"True"`:

https://github.com/saltstack/salt/blob/20d5073a9186439b08f63e49b96c6d789ba6bda7/salt/modules/solarispkg.py#L359-L361

This should be a boolean test.

**Expected behavior**

`current_zone_only=True` should work.

**Versions Report**

This is a second-hand report against 3000.4, but it appears to still be an issue on `master`.

**Additional context**

ZD-6213.

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Allows for a Boolean value for 'current_zone_only' when performing a pkg.install on Solaris 10, as per documentation

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/59305

### Previous Behavior
Boolean values for kwarg current_zone_only were ignored, only checked if text was "True".

### New Behavior
Allows for kwarg current_zone_only to be a Boolean value, or a text value of "True" (to not break anything installed in the field)

Also added unit module pytest for the functionality.
Note: no previous tests were found for Solaris 10 pkg.install functionality

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/solarispkg.py [hunk 1]
```diff
@@ -360,7 +360,10 @@ def install(name=None, sources=None, saltenv="base", **kwargs):
         cmd_prefix = ["/usr/sbin/pkgadd", "-n", "-a", adminfile]
 
         # Only makes sense in a global zone but works fine in non-globals.
-        if kwargs.get("current_zone_only") == "True":
+        if (
+            kwargs.get("current_zone_only") is True
+            or kwargs.get("current_zone_only") == "True"
+        ):
```
Comment (s0undt3ch) [inline salt/modules/solarispkg.py:366]:
```suggestion
        if kwargs.get("current_zone_only") in (True, "True"):
```

Comment (s0undt3ch) [reply salt/modules/solarispkg.py:366]: Or even:
```suggestion
        if str(kwargs.get("current_zone_only")) == "True"):
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/modules/solarispkg.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/modules/solarispkg.py:366]: I'd say the first suggestion would be the more correct

Comment (dmurphy18) [reply salt/modules/solarispkg.py]: Did the first suggestion

**Review by s0undt3ch — APPROVED** [review]

**Review by krionbsd — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**